### ISSUE TITLE
eval.py exception

### ISSUE THREAD

**Issue by BonjourGit** [nokia/nlc2cmd-submission-hubris#1]
Hi,

Python 3.7.9, with all the requirements being installed.
However, eval.py throws exception:

```
# python eval.py test/dev_nl.txt
[WARNING]: Importing of metric computation failed. This may somewhat decrease performance.
-- Loading models...
Downloading: 100%|███████████████████████████████████████████████████████████| 1.04M/1.04M [00:02<00:00, 436kB/s]
Downloading: 100%|█████████████████████████████████████████████████████████████| 456k/456k [00:01<00:00, 278kB/s]
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/transformers/configuration_utils.py", line 242, in get_config_dict
    raise EnvironmentError
OSError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "eval.py", line 189, in <module>
    main()
  File "eval.py", line 164, in main
    model1 = get_model('jaron-maene/gpt2-large-nl2bash')
  File "eval.py", line 78, in get_model
    return transformers.GPT2LMHeadModel.from_pretrained(f'{model_name}')
  File "/usr/local/lib/python3.7/site-packages/transformers/modeling_utils.py", line 604, in from_pretrained
    **kwargs,
  File "/usr/local/lib/python3.7/site-packages/transformers/configuration_utils.py", line 200, in from_pretrained
    config_dict, kwargs = cls.get_config_dict(pretrained_model_name_or_path, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/transformers/configuration_utils.py", line 251, in get_config_dict
    raise EnvironmentError(msg)
OSError: Can't load config for 'jaron-maene/gpt2-large-nl2bash'. Make sure that:

- 'jaron-maene/gpt2-large-nl2bash' is a correct model identifier listed on 'https://huggingface.co/models'

- or 'jaron-maene/gpt2-large-nl2bash' is the correct path to a directory containing a config.json file

```
Also, cannot import 'data_tools' from 'bashlint'. 
```
# python
Python 3.7.9 (default, Dec 22 2020, 08:45:09)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from src.tm_metric import compute_metric
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/root/github/nlc2cmd-submission-hubris/src/tm_metric.py", line 16, in <module>
    from bashlint import data_tools, nast
ImportError: cannot import name 'data_tools' from 'bashlint' (/usr/local/lib/python3.7/site-packages/bashlint.py)
>>>
```

**Comment by jjcmoon** [comment]
Strange, I don't really know why it fails to download the model. Never had that issue. I just tried to load the model on [a google colab](https://colab.research.google.com/drive/1ZUaPm6yJyi-KvNpDrQV71JkXHYX_sr-y?usp=sharing) where it worked as expected. Maybe you can try with the same versions used here (python 3.6.9 and transformers 4.1.1)?

Concerning the failed bashlint import. this was purposefully not included in the public repo. You can safely ignore this. It has no impact at all if you're only interested in top1 predictions, and only very slight impact for top5.

**Comment by BonjourGit** [comment]
Thanks.

Trying with transformers 4.1.1. It seems now it is trying to download the models.

```
# python eval.py test/dev_nl.txt
In Transformers v4.0.0, the default path to cache downloaded models changed from '~/.cache/torch/transformers' to '~/.cache/huggingface/transformers'. Since you don't seem to have overridden and '~/.cache/torch/transformers' is a directory that exists, we're moving it to '~/.cache/huggingface/transformers' to avoid redownloading models you have already in the cache. You should only see this message once.
/usr/local/lib/python3.7/site-packages/torch/cuda/__init__.py:52: UserWarning: CUDA initialization: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx (Triggered internally at  /pytorch/c10/cuda/CUDAFunctions.cpp:100.)
  return torch._C._cuda_getDeviceCount() > 0
[WARNING]: Importing of metric computation failed. This may somewhat decrease performance.
-- Loading models...
Downloading: 100%|██████████████████████████| 1.04M/1.04M [00:02<00:00, 413kB/s]
Downloading: 100%|████████████████████████████| 456k/456k [00:01<00:00, 382kB/s]
Downloading: 100%|██████████████████████████████| 765/765 [00:00<00:00, 293kB/s]
Downloading:  43%|██████████▎             | 1.35G/3.13G [45:18<1:00:12, 495kB/s]
```

Existing requirements.txt specifies 3.0.2 though:
transformers==3.0.2

**Comment by BonjourGit** [comment]
Interestingly, the model predicts different output for the same input for repeated runs, and same output for different input:

=== 1st run ===
```
# python eval.py test/dev_nl.txt
/usr/local/lib/python3.7/site-packages/torch/cuda/__init__.py:52: UserWarning: CUDA initialization: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx (Triggered internally at  /pytorch/c10/cuda/CUDAFunctions.cpp:100.)
  return torch._C._cuda_getDeviceCount() > 0
[WARNING]: Importing of metric computation failed. This may somewhat decrease performance.
-- Loading models...
Downloading: 100%|██████████████████████████| 1.04M/1.04M [00:02<00:00, 475kB/s]
Downloading: 100%|████████████████████████████| 456k/456k [00:01<00:00, 393kB/s]
-- Models loaded
-- Starting prediction
invocation:
                list files in current directory
results:
                ls File
                ls | awk Program
                find Path -type f | xargs -I {} ls {}
                find Path -name Regex -printf '%h\n'
                find Path -ls
```
   
=== 2nd run ===
```
# python eval.py test/dev_nl.txt
/usr/local/lib/python3.7/site-packages/torch/cuda/__init__.py:52: UserWarning: CUDA initialization: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx (Triggered internally at  /pytorch/c10/cuda/CUDAFunctions.cpp:100.)
  return torch._C._cuda_getDeviceCount() > 0
[WARNING]: Importing of metric computation failed. This may somewhat decrease performance.
-- Loading models...
-- Models loaded
-- Starting prediction
invocation:
                list files in current directory
results:
                find Path | xargs -n Quantity -I {} tar -c -f File {}
                find Path | xargs -I {} tar -c -f File {}
                find Path | xargs -I {} tar -c -z -f File {}
                find Path -type f -name Regex | awk Program | xargs -I {} tar -c -f File {}
                find Path -type f -exec tar -c -f File {} \;
-- Starting prediction
invocation:
                find all filenames including ntas in current directory and create a tarball
results:
                find Path | xargs -n Quantity -I {} tar -c -f File {}
                find Path | xargs -I {} tar -c -f File {}
                find Path | xargs -I {} tar -c -z -f File {}
                find Path -type f -name Regex | awk Program | xargs -I {} tar -c -f File {}
                find Path -type f -exec tar -c -f File {} \;
```
   
=== 3rd run ===
```
# python eval.py test/dev_nl.txt
/usr/local/lib/python3.7/site-packages/torch/cuda/__init__.py:52: UserWarning: CUDA initialization: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx (Triggered internally at  /pytorch/c10/cuda/CUDAFunctions.cpp:100.)
  return torch._C._cuda_getDeviceCount() > 0
[WARNING]: Importing of metric computation failed. This may somewhat decrease performance.
-- Loading models...
-- Models loaded
-- Starting prediction
invocation:
                list files in current directory
results:
                find Path -type f | xargs -I {} ls -l -d {} | tail -n Quantity
                find Path \(-name Regex -or -name Regex -or -name Regex \) | xargs -I {} echo Regex Regex Regex {}
                find Path | xargs -I {} echo Regex Regex Regex {}
                find Path -name Regex -exec echo {} \; | awk Program
                find Path \(-name Regex -or -name Regex -or -name Regex \) -ls
-- Starting prediction
invocation:
                show the linux kernel version
results:
                find Path -type f | xargs -I {} ls -l -d {} | tail -n Quantity
                find Path \(-name Regex -or -name Regex -or -name Regex \) | xargs -I {} echo Regex Regex Regex {}
                find Path | xargs -I {} echo Regex Regex Regex {}
                find Path -name Regex -exec echo {} \; | awk Program
                find Path \(-name Regex -or -name Regex -or -name Regex \) -ls
```
   
Input file under test:
```
# cat test/dev_nl.txt
list files in current directory
show linux kernel version
```

**Comment by jjcmoon** [comment]
My apologies, there was a small typo in the eval script. But I see you already figured that out yourself.
For the transformers version, I declared the older version as it was the one I used during development. For some of the experiments (like the diverse beam search) I monkey patched some internal transformers functions, so I suspect those might be broken when using the new version.
That being said, this is much less important than just being able to download the models (which for some reason doesn't work anymore with older versions), so I have accepted your pull request.
